src/domain/bitcoin/lightning/ln-invoice.ts

```diff
@@ -56,7 +56,7 @@ export const decodeInvoice = (
     description: decodedInvoice.description || "",
     paymentHash: decodedInvoice.id as PaymentHash,
     destination: decodedInvoice.destination as Pubkey,
-    milliSatsAmount: toMilliSatsFromNumber(decodedInvoice.mtokens || 0),
+    milliSatsAmount: toMilliSatsFromNumber(amount ? amount * 1000 : 0),
```

Comment:
this is required to avoid `ExpectedEqualValuesForTokensAndMtokens` exception from lightning library